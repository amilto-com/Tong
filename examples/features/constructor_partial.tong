// Demonstrate partial application of multi-arg constructors

data Tree = Leaf v | Node left right

// Create a partial by applying only the left subtree
let left_partial = Node(Leaf(1))         // <partial:Node:1>
let full = left_partial(Leaf(2))         // Node(Leaf(1),Leaf(2))
print(full)

// Another example
let p = Node(Leaf(3))
let p2 = p(Leaf(4))
print(p2)
