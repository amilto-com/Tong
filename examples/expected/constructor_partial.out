Node(Leaf(1),Leaf(2)) Node(Leaf(3),Leaf(4))